[botan] build failure - Unbound var implib_name

### Operating system

Linux (x64-linux-dynamic)

### Compiler

GCC

### Steps to reproduce the behavior

```Shell
Try to install botan package with x64-linux-dynamic triplet.
```


### Failure logs

("dirname" replaced real path in logs)
```
Building botan:x64-linux-dynamic@3.3.0...
<dirname>/vcpkg/triplets/community/x64-linux-dynamic.cmake: info: loaded community triplet from here. Community triplets are not built in the curated registry and are thus less likely to succeed.
-- Downloading https://github.com/randombit/botan/archive/3.3.0.tar.gz -> randombit-botan-3.3.0.tar.gz...
-- Extracting source <dirname>/vcpkg/downloads/randombit-botan-3.3.0.tar.gz
-- Applying patch embed-debug-info.patch
-- Applying patch pkgconfig.patch
-- Applying patch verbose-install.patch
-- Applying patch configure-zlib.patch
-- Applying patch fix_android.patch
-- Applying patch libcxx-winpthread-fixes.patch
-- Applying patch fix-cmake-usage.patch
-- Using source at <dirname>/vcpkg/buildtrees/botan/src/3.3.0-c527fba307.clean
-- Getting CMake variables for x64-linux-dynamic
-- Getting CMake variables for x64-linux-dynamic-dbg
-- Getting CMake variables for x64-linux-dynamic-rel
-- Configuring x64-linux-dynamic-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /usr/bin/bash -c "V=1 ./../src/3.3.0-c527fba307.clean/configure  \"/usr/bin/python3\" \"<dirname>/vcpkg/buildtrees/botan/src/3.3.0-c527fba307.clean/configure.py\" \"--distribution-info=vcpkg x64-linux-dynamic\" \"--disable-cc-tests\" \"--with-pkg-config\" \"--link-method=copy\" \"--with-debug-info\" \"--includedir=include\" \"--bindir=bin\" \"--libdir=lib\" \"--without-documentation\" \"--with-external-includedir=<dirname>/vcpkg/installed/x64-linux-dynamic/include\" \"--cpu=x86_64\" \"--cc=gcc\" \"--enable-shared\" \"--disable-static\" \"--debug-mode\" \"--prefix=<dirname>/vcpkg/packages/botan_x64-linux-dynamic/debug\" \"--with-external-libdir=<dirname>/vcpkg/installed/x64-linux-dynamic/debug/lib\" \"--prefix=<dirname>/vcpkg/installed/x64-linux-dynamic/debug\""
    Working Directory: <dirname>/vcpkg/buildtrees/botan/x64-linux-dynamic-dbg
    Error code: 1
    See logs for more information:
      <dirname>/vcpkg/buildtrees/botan/config-x64-linux-dynamic-dbg-out.log
      <dirname>/vcpkg/buildtrees/botan/config-x64-linux-dynamic-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:863 (vcpkg_execute_required_process)
  ports/botan/portfile.cmake:154 (vcpkg_configure_make)
  scripts/ports.cmake:191 (include)
```
```
   INFO: <dirname>/vcpkg/buildtrees/botan/src/3.3.0-c527fba307.clean/configure.py invoked with options "--distribution-info=vcpkg x64-linux-dynamic --disable-cc-tests --with-pkg-config --link-method=copy --with-debug-info --includedir=include --bindir=bin --libdir=lib --without-documentation --with-external-includedir=<dirname>/vcpkg/installed/x64-linux-dynamic/include --cpu=x86_64 --cc=gcc --enable-shared --disable-static --debug-mode --prefix=<dirname>/vcpkg/packages/botan_x64-linux-dynamic/debug --with-external-libdir=<dirname>/vcpkg/installed/x64-linux-dynamic/debug/lib"
   INFO: Configuring to build Botan 3.3.0 (revision unknown)
   INFO: Python version: "3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]"
   INFO: CXX environment variable is set which will override compiler path
   INFO: Implicit --ar-command=/usr/bin/ar due to environment variable AR
   INFO: Implicit --cc-bin=/usr/bin/c++ due to environment variable CXX
   INFO: Implicit --cxxflags= -fPIC -g due to environment variable CXXFLAGS
   INFO: Implicit --ldflags=-L<dirname>/vcpkg/installed/x64-linux-dynamic/debug/lib due to environment variable LDFLAGS
   INFO: Autodetected platform information: OS="Linux" machine="x86_64" proc=""
   INFO: Guessing target OS is linux (use --os to set)
   INFO: Using /etc/ssl/certs/ca-certificates.crt as system certificate store
   INFO: Target is gcc:0.0-linux-x86_64
   INFO: Assuming target x86_64 is little endian
   INFO: Skipping (dependency failure): asio certstor_sqlite3 sessions_sqlite3
   INFO: Skipping (incompatible CPU): aes_armv8 aes_power8 sha1_armv8 sha2_32_armv8 sha2_64_armv8 shacal2_armv8 sm4_armv8
   INFO: Skipping (incompatible OS): certstor_system_macos certstor_system_windows commoncrypto win32_stats
   INFO: Skipping (no enabled compression schemes): compression
   INFO: Skipping (requires external dependency): boost bzip2 lzma sqlite3 tpm zlib
   INFO: Loading modules: adler32 aead aes aes_crystals_xof aes_ni aes_vperm argon2 argon2_avx2 argon2_ssse3 argon2fmt aria asn1 auto_rng base base32 base58 base64 bcrypt bcrypt_pbkdf bigint blake2 blake2mac blake2s block blowfish camellia cascade cast128 cbc ccm certstor_flatfile certstor_sql certstor_system cfb chacha chacha20poly1305 chacha_avx2 chacha_avx512 chacha_rng chacha_simd32 checksum cmac comb4p cpuid crc24 crc32 cryptobox cshake_xof ctr curve25519 des dh dilithium dilithium_aes dilithium_common dl_algo dl_group dlies dsa dyn_load eax ec_group ec_h2c ecc_key ecdh ecdsa ecgdsa ecies eckcdsa ed25519 elgamal eme_oaep eme_pkcs1 eme_raw emsa_pkcs1 emsa_pssr emsa_raw emsa_x931 entropy fd_unix ffi filters fpe_fe1 frodokem frodokem_aes frodokem_common gcm getentropy ghash ghash_cpu ghash_vperm gmac gost_28147 gost_3410 gost_3411 hash hash_id hex hkdf hmac hmac_drbg hotp http_util idea idea_sse2 iso9796 kdf kdf1 kdf1_iso18033 kdf2 keccak keccak_perm keccak_perm_bmi2 keypair kmac kuznyechik kyber kyber_90s kyber_common lion locking_allocator mac mce md4 md5 mdx_hash mem_pool mgf1 mode_pad modes mp nist_keywrap noekeon noekeon_simd numbertheory ocb ofb par_hash passhash9 pbes2 pbkdf pbkdf2 pem pgp_s2k pk_pad pkcs11 poly1305 poly_dbl prf_tls prf_x942 processor_rng psk_db pubkey raw_hash rc4 rdseed rfc3394 rfc6979 rmd160 rng roughtime rsa salsa20 scrypt seed serpent serpent_avx2 serpent_avx512 serpent_simd sessions_sql sha1 sha1_sse2 sha1_x86 sha2_32 sha2_32_bmi2 sha2_32_x86 sha2_64 sha2_64_bmi2 sha3 shacal2 shacal2_avx2 shacal2_simd shacal2_x86 shake shake_cipher shake_xof simd simd_avx2 simd_avx512 siphash siv skein sm2 sm3 sm4 socket sodium sp800_108 sp800_56a sp800_56c sphincsplus_common sphincsplus_sha2 sphincsplus_shake srp6 stateful_rng stream streebog system_rng thread_utils threefish_512 tls tls12 tls13 tls13_pqc tls_cbc trunc_hash tss twofish utils uuid whirlpool x509 x919_mac xmss xof xts zfec zfec_sse2 zfec_vperm
  ERROR: Unbound var 'implib_name' in template <dirname>/buildtrees/botan/src/3.3.0-c527fba307.clean/src/build-data/botan-config.cmake.in
```

### Additional context

_No response_

--- a/ports/botan/fix-cmake-usage.patch
+++ b/ports/botan/fix-cmake-usage.patch
@@ -1,8 +1,8 @@
 diff --git a/src/build-data/botan-config.cmake.in b/src/build-data/botan-config.cmake.in
-index 8d14c4e..46e2cbc 100644
+index 8d14c4e..cfc9277 100644
 --- a/src/build-data/botan-config.cmake.in
 +++ b/src/build-data/botan-config.cmake.in
-@@ -65,21 +65,29 @@ if(DEFINED ${CMAKE_FIND_PACKAGE_NAME}_FOUND AND NOT ${${CMAKE_FIND_PACKAGE_NAME}
+@@ -65,27 +65,36 @@ if(DEFINED ${CMAKE_FIND_PACKAGE_NAME}_FOUND AND NOT ${${CMAKE_FIND_PACKAGE_NAME}
    return()
  endif()
  
@@ -36,7 +36,16 @@ index 8d14c4e..46e2cbc 100644
  endif()
  %{endif}
  
-@@ -100,10 +108,20 @@ if(NOT TARGET Botan::Botan)
+ %{if implib_name}
+-set(_Botan_implib     "${_Botan_PREFIX}/lib/%{implib_name}")
+-set(_Botan_shared_lib "${_Botan_PREFIX}/bin/%{shared_lib_name}")
++set(_Botan_implib         "${_Botan_PREFIX}/lib/%{implib_name}")
++set(_Botan_implib_debug   "${_Botan_PREFIX}/debug/lib/%{implib_name}")
++set(_Botan_shared_lib     "${_Botan_PREFIX}/bin/%{shared_lib_name}")
+ %{endif}
+ %{unless implib_name}
+ set(_Botan_implib "")
+@@ -100,10 +109,28 @@ if(NOT TARGET Botan::Botan)
    add_library(Botan::Botan SHARED IMPORTED)
    set_target_properties(Botan::Botan
      PROPERTIES
@@ -45,18 +54,26 @@ index 8d14c4e..46e2cbc 100644
 -      INTERFACE_INCLUDE_DIRECTORIES "${_Botan_PREFIX}/include/botan-%{version_major}"
 +      INTERFACE_INCLUDE_DIRECTORIES "${_Botan_PREFIX}/include"
        INTERFACE_LINK_OPTIONS        "SHELL:%{cxx_abi_flags}")
-+  if(EXISTS "${_Botan_PREFIX}/debug/lib/%{implib_name}")
++  if(EXISTS "${_Botan_PREFIX}/debug/lib/%{shared_lib_name}")
 +    set_property(TARGET Botan::Botan APPEND PROPERTY IMPORTED_CONFIGURATIONS DEBUG)
 +    set_target_properties(Botan::Botan PROPERTIES
-+      IMPORTED_IMPLIB_DEBUG "${_Botan_PREFIX}/debug/lib/%{implib_name}"
 +      IMPORTED_LOCATION_DEBUG "${_Botan_PREFIX}/debug/bin/%{shared_lib_name}"
 +    )
++    if(NOT _Botan_implib STREQUAL "")
++      set_target_properties(Botan::Botan PROPERTIES
++        IMPORTED_IMPLIB_DEBUG ${_Botan_implib_debug}
++      )
++    endif()
 +  endif()
 +  set_property(TARGET Botan::Botan APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
 +  set_target_properties(Botan::Botan PROPERTIES
-+    IMPORTED_IMPLIB_RELEASE "${_Botan_PREFIX}/lib/%{implib_name}"
 +    IMPORTED_LOCATION_RELEASE "${_Botan_PREFIX}/bin/%{shared_lib_name}"
++  )
++  if(NOT _Botan_implib STREQUAL "")
++    set_target_properties(Botan::Botan PROPERTIES
++      IMPORTED_IMPLIB_RELEASE ${_Botan_implib}
 +    )
++  endif()
    set_property(TARGET Botan::Botan APPEND PROPERTY IMPORTED_CONFIGURATIONS NOCONFIG)
    set_target_properties(Botan::Botan
      PROPERTIES

--- a/ports/botan/vcpkg.json
+++ b/ports/botan/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "botan",
   "version": "3.3.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "A cryptography library written in C++11",
   "homepage": "https://botan.randombit.net",
   "license": "BSD-2-Clause",

--- a/versions/b-/botan.json
+++ b/versions/b-/botan.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "24b20ce86f1fff75cd0c14c71e8ddcd7459b48c8",
+      "version": "3.3.0",
+      "port-version": 2
+    },
+    {
       "git-tree": "4e721d4e37cd859a5134c2cbe587aed5644bc1de",
       "version": "3.3.0",
       "port-version": 1

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -1350,7 +1350,7 @@
     },
     "botan": {
       "baseline": "3.3.0",
-      "port-version": 1
+      "port-version": 2
     },
     "box2d": {
       "baseline": "2.4.1",

